toolkits/Dockerfile.ml

```diff
@@ -0,0 +1,221 @@
+ARG CPP_DEPS=/tmp/cpp_deps
+ARG PY_DEPS=/tmp/py_deps
+ARG PYTHON_VERSION=3.12
+ARG UBUNTU_VERSION=24.04
+
+ARG TRT_IMAGE_TAG=24.12-py3
+ARG TENSORRT_IMAGE=nvcr.io/nvidia/tensorrt:${TRT_IMAGE_TAG}
+
+# TODO: newer versions have issues with eigen3 / main works, but we need a tag (use tag right after v1.22.1 when available)
+ARG ONNX_RUNTIME_VERSION=main 
+
+FROM ubuntu:${UBUNTU_VERSION} AS onnx-source
+
+ARG ONNX_RUNTIME_VERSION
+RUN apt-get update \
+ && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    ca-certificates \
+    git
+
+WORKDIR /tmp/onnxruntime
+RUN git clone -b ${ONNX_RUNTIME_VERSION} --recursive https://github.com/microsoft/onnxruntime . \
+ # TODO: remove this when we have a tag for v1.22.1+ \
+ && git checkout 4a730ca \
+ && git submodule update --init --recursive
+
+FROM ubuntu:${UBUNTU_VERSION} AS cpu-cpp-builder
+
+ARG CPP_DEPS
+ARG PY_DEPS
+ARG ONNX_RUNTIME_VERSION
+ARG USE_TENSORRT=OFF
+ARG USE_TENSORRT_BUILTIN_PARSER=OFF
+
+RUN apt-get update \
+ && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    ca-certificates \
+    build-essential \
+    cmake \
+    git \
+    wget \
+    python3-pip \
+    python3-pybind11 \
+    python3-dev \
+    libssl-dev \
+    libffi-dev \
+    flatbuffers-compiler \
+    libeigen3-dev \
+  && update-ca-certificates \
+  && rm -rf /var/lib/apt/lists/* \
+  && mkdir -p ${CPP_DEPS}/lib ${CPP_DEPS}/include
+RUN pip install --no-cache-dir --break-system-packages \
+  flatbuffers \
+  packaging \
+  numpy==1.26.4 
+
+WORKDIR /tmp
+# install ONNX runtime library
+COPY --from=onnx-source /tmp/onnxruntime/ /tmp/onnxruntime
+WORKDIR /tmp/onnxruntime
+RUN ./build.sh \
+  --config Release \
+  --build_shared_lib \
+  --build_wheel \
+  --parallel \
+  --skip_tests \
+  --compile_no_warning_as_error \
+  --skip_submodule_sync \
+  --enable_pybind \
+  --allow_running_as_root
+RUN cmake --install build/Linux/Release --prefix ${CPP_DEPS}
+# build python package too
+RUN mkdir -p ${PY_DEPS} \ 
+  && cd build/Linux/Release/dist/ \
+  && pip install --no-cache-dir \
+      --target=${PY_DEPS} \
+      --no-deps \
+      onnxruntime*.whl
+
+FROM ${TENSORRT_IMAGE} AS cuda-cpp-builder
+ARG CPP_DEPS ONNX_RUNTIME_VERSION
+
+ARG CPP_DEPS
+ARG PY_DEPS
+ARG ONNX_RUNTIME_VERSION
+ARG USE_CUDA=ON
+ARG USE_TENSORRT=ON
+ARG USE_TENSORRT_BUILTIN_PARSER=OFF
+ARG CMAKE_VERSION=3.28
+ARG CMAKE_BUILD=2
+
+RUN apt-get update \
+ && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    ca-certificates \
+    build-essential \
+    cmake \
+    git \
+    wget \
+    python3-pip \
+    python3-pybind11 \
+    python3-dev \
+    libssl-dev \
+    libffi-dev \
+    flatbuffers-compiler \
+    libeigen3-dev \
+    tar \
+  && rm -rf /var/lib/apt/lists/* \
+  && update-ca-certificates \
+  && mkdir -p ${CPP_DEPS}/lib ${CPP_DEPS}/include
```

Comment:
I could probably partially copy apt deps from another stage, but at this version of the image I don't think it will have a big impact